controllers/storagecluster/cephcluster.go

```diff
@@ -1559,3 +1559,54 @@ func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string)
 
 	return crs
 }
+
+// setMonTargetPgPerOsd sets the mon_target_pg_per_osd value to 400 if not already set
+func setMonTargetPgPerOsd(cephConfig *map[string]map[string]string) {
+	if *cephConfig == nil {
+		*cephConfig = make(map[string]map[string]string)
+	}
+	if global, exists := (*cephConfig)["global"]; exists {
+		if _, exists := global["mon_target_pg_per_osd"]; !exists {
+			global["mon_target_pg_per_osd"] = "400"
+		}
+	} else {
+		(*cephConfig)["global"] = map[string]string{
+			"mon_target_pg_per_osd": "400",
+		}
+	}
+}
+
+// preserveMonTargetPgPerOsd preserves the mon_target_pg_per_osd value if its present in the existing cephconfig but is missing in the updated cephconfig
+func preserveMonTargetPgPerOsd(existingCephConfig map[string]map[string]string, updatedCephConfig *map[string]map[string]string) {
+	if global, exists := existingCephConfig["global"]; exists {
+		if value, exists := global["mon_target_pg_per_osd"]; exists {
+			if *updatedCephConfig == nil {
+				*updatedCephConfig = make(map[string]map[string]string)
+			}
+			if _, exists := (*updatedCephConfig)["global"]; !exists {
+				(*updatedCephConfig)["global"] = make(map[string]string)
+			}
+			(*updatedCephConfig)["global"]["mon_target_pg_per_osd"] = value
+		}
+	}
+}
+
+func getCephClusterCephConfig(sc *ocsv1.StorageCluster) map[string]map[string]string {
+	cephConfig := map[string]map[string]string{
+		"global": {
+			"mon_max_pg_per_osd": "1000",
+		},
```

Comment:
Can we include the `mon_target_pg_per_osd` in this method, then no need for the separate methods for that specific setting? I'm hoping this implementation will be able to simply handle any default settings we want in the cephconfig, and also allow the user to override any of them if desired. I don't think there will be a need to treat any of these settings differently.

Reply:
We want to not apply the mon_target_pg_per_osd setting to upgraded clusters, setting it here would make it apply to upgraded clusters as well.